### PR TITLE
332/update affiliate flow

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Part of #332 

Updating affiliate flow to use new metadata handling

# To Test

Test affiliate flow as usual.
Should work same way as before, except that IPFS file is not uploaded when following the link

Use the same test steps from https://github.com/cowprotocol/cowswap/pull/629 to check on the console logs to get and check generated metadata includes the referral address

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @alfetopito , great job!

The only issue I have faced is that there is no wrong affiliate code message when the code is incorrect
invalid code
штм

All the rest issues I have faced are regression ones: 
 - https://github.com/cowprotocol/cowswap/issues/679
 - https://github.com/cowprotocol/cowswap/issues/680
 - https://github.com/cowprotocol/cowswap/issues/204#issuecomment-1154011255

Thanks!

**Comment by elena-zh** [comment]
Another issue I have faced is that referrals section is not updated now: I do not see changes for the referral link owner `0x1f5B740436Fc5935622e92aa3b46818906F416E9`

Total trades for the folllower referral link account `0x068C7bb435Ab1701957Ea9a8757485EAf4cCF731 `are updated
image

**Comment by elena-zh** [comment]
I have also noticed that Prod API does not show a trade for the account 
missed trade

File: src/custom/state/appData/utils.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { AppDataKeyParams } from 'state/appData/types'
+
+export function buildAppDataRecordKey({ chainId, orderId }: AppDataKeyParams): string {
+  return `${chainId}-${orderId}`
+}
+
+export function parseAppDataRecordKey(key: string): AppDataKeyParams {
+  const [chainId, orderId] = key.split('-')
+
+  return { chainId, orderId } as unknown as AppDataKeyParams
```
Comment (W3stside) [inline src/custom/state/appData/utils.ts:10]: what's the complaint from TS here?

File: src/custom/state/affiliate/hooks.ts [hunk 2]
```diff
@@ -1,33 +1,13 @@
...
-        isValid: boolean
-      }
-    | undefined
-  >((state) => {
-    return state.affiliate.referralAddress
-  })
+  return useAppSelector((state) => state.affiliate.referralAddress)
 }
 
 export function useAddress() {
```
Comment (anxolin) [inline src/custom/state/affiliate/hooks.ts:9]: not from this PR, but shouldn't the name be `useAffiliateAddress`? more coherent with `useReferralAddress` and also when imported this hook, will be a too general name in whichever component (so the reader won't related it to the affiliate program)

**Review by anxolin — APPROVED** [review]
Code looks good. U would rely on Elena on double-checking the IPFS document that is created. 

If you manage to create in production a trade for a brand new account that followed a link, and you share with me the order in the explorer, I can verify the document is correct.

File: src/custom/state/appData/updater.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { useAtomValue, useSetAtom } from 'jotai'
+import { useEffect, useRef } from 'react'
+import ms from 'ms.macro'
+
+import { COW_SDK } from 'constants/index'
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/state/appData/updater.ts]: What would be a better place for it?

**Comment by elena-zh** [comment]
> I have also noticed that Prod API does not show a trade for the account missed trade

No longer reproducible: the data is displayed in both in Prod and in Barn

**Comment by elena-zh** [comment]
> Another issue I have faced is that referrals section is not updated now: I do not see changes for the referral link owner `0x1f5B740436Fc5935622e92aa3b46818906F416E9`
> 
> Total trades for the follower referral link account `0x068C7bb435Ab1701957Ea9a8757485EAf4cCF731 `are updated image

This issue seems to be still relevant. 
See: the data on Prod shows a correct number of referrals trades/volume, but in the PR the last trade done during tests  (1 trade with 16$ volume) is not included
image

Order info that is not included into referrals: https://protocol-explorer.dev.gnosisdev.com/orders/0x0b20c7803e91af6ab9f7297e0a8737d625e92f3f4403aab8c60c577de5a71911068c7bb435ab1701957ea9a8757485eaf4ccf73162a76194

**Comment by alfetopito** [comment]
> Hey @alfetopito , great job!
> 
> The only issue I have faced is that there is no wrong affiliate code message when the code is incorrect invalid code штм

This should be fixed now

---

As for the data discrepancy (5 referrals on barn vs 6 on prod), I'm checking with the backend.

File: src/custom/state/affiliate/hooks.ts (hunk 2 above)
Comment (alfetopito) [reply src/custom/state/affiliate/hooks.ts:9]: Will address in a follow up PR

**Comment by elena-zh** [comment]
Changes related to an invalid affiliate code LGTM

**Comment by elena-zh** [comment]
Hey @alfetopito , I have not verified an issue `(5 referrals on barn vs 6 on prod)` yet. Is it fixed? 
Should I verify it inside #629 ?

File: src/custom/state/affiliate/hooks.ts (hunk 2 above)
Comment (alfetopito) [reply src/custom/state/affiliate/hooks.ts:9]: https://github.com/cowprotocol/cowswap/pull/748

**Comment by alfetopito** [comment]
> Hey @alfetopito , I have not verified an issue `(5 referrals on barn vs 6 on prod)` yet. Is it fixed? Should I verify it inside #629 ?

Alex confirmed it has been fixed, it was an issue with the backend.

You can check by looking at the response from both APIs:

https://barn.api.cow.fi/affiliate/api/v1/profile/0x1f5B740436Fc5935622e92aa3b46818906F416E9
https://api.cow.fi/affiliate/api/v1/profile/0x1f5B740436Fc5935622e92aa3b46818906F416E9

**Pull request merged**